libraries/SITL/SIM_AirSim.cpp

```diff
@@ -0,0 +1,158 @@
+/* 
+	Simulator Connector for AirSim
+*/
+
+#include "SIM_AirSim.h"
+
+#include <stdio.h>
+#include <arpa/inet.h>
+#include <errno.h>
+
+#include <AP_HAL/AP_HAL.h>
+
+extern const AP_HAL::HAL& hal;
+
+using namespace SITL;
+
+AirSim::AirSim(const char *home_str, const char *frame_str) :
+	Aircraft(home_str, frame_str),
+	sensor_sock(true),
+	control_sock(true),
+	last_timestamp(0),
+	opened_control_socket(false),
+	opened_sensor_socket(false)
+{
+	printf("Starting Airsim at %s:%u:%u\n", airsim_ip, airsim_sensor_port, airsim_control_port);
+}
+
+/*
+	Connect to required sockets
+*/
+bool AirSim::connect_sensor_socket(void)
+{
+	if (opened_sensor_socket) {
+		return true;
+	}
+	// This should be binding to 127.0.0.1 or 0.0.0.0 ?
+	if(!sensor_sock.bind(airsim_ip, airsim_sensor_port)) {
+		printf("Unable to open Airsim sensor socket at %s:%u\n", airsim_ip, airsim_sensor_port);
+		printf("Error: %s\n", strerror(errno));
+		return false;
+	}
+	printf("Opened Airsim sensor socket at %s:%u\n", airsim_ip, airsim_sensor_port);
+	sensor_sock.set_blocking(false);
+	sensor_sock.reuseaddress();
+	opened_sensor_socket = true;
+
+	return true;
+}
+
+bool AirSim::connect_control_socket(void)
+{
+	if (opened_control_socket) {
+		return true;
+	}
+	// binding to 0.0.0.0 or 127.0.0.1 ?
+	if(!control_sock.bind(airsim_ip, airsim_control_port)) {
+		printf("Unable to open Airsim control socket at %s:%u\n", airsim_ip, airsim_control_port);
+		printf("Error: %s\n", strerror(errno));
+		return false;
+	}
+	printf("Opened Airsim control socket at %s:%u\n", airsim_ip, airsim_control_port);
+	control_sock.set_blocking(false);
+	control_sock.reuseaddress();
+	opened_control_socket = true;
+
+	return true;
+}
+
+void AirSim::send_servos(const struct sitl_input &input)
+{
+	servo_packet pkt;
+
+	for(int i=0; i<kArduCopterRotorControlCount; ++i) {
```

Comment:
We use `uint8_t` for loop counters in ardupilot and always post-increment rather than pre-increment.  That's a style thing (and the google style guide says the opposite... :-) )

Reply:
Done

Reply:
Should this be added to the style-guide?